clarifai/rag/utils.py

```diff
@@ -26,3 +33,71 @@ def convert_messages_to_str(messages: List[dict]) -> str:
 
 def format_assistant_message(raw_text: str) -> dict:
   return {"role": "assistant", "content": raw_text}
+
+
+def load_documents(file_path: str = None, folder_path: str = None,
+                   url: str = None) -> List[Document]:
+  """Loads documents from a local directory or public url or local filename.
+
+  Args:
+      file_path (str): The path to the filename.
+      folder_path (str): The path to the folder.
+      url (str): The url to the file.
+  """
+  #document loaders for filepath
+  if file_path:
+    if file_path.endswith(".pdf"):
+      PDFReader = download_loader("PDFReader")
+      loader = PDFReader()
+      documents = loader.load_data(file=Path(file_path))
+    elif file_path.endswith(".docx"):
+      docReader = download_loader("DocxReader")
+      loader = docReader()
+      documents = loader.load_data(file=Path(file_path))
+    elif file_path.endswith(".txt"):
+      with open(file_path, 'r') as file:
+        text_content = file.read()
+      documents = [Document(text=text_content)]
+    else:
+      raise ValueError("Only .pdf, .docx, and .txt files are supported.")
+
+  #document loaders for folderpath
+  if folder_path:
+    documents = SimpleDirectoryReader(
+        input_dir=Path(folder_path), required_exts=[".pdf", ".docx"]).load_data()
+
+  #document loaders for url
+  if url:
+    response = requests.get(url)
+    if response.status_code == 200:
```

Comment:
Better to unnest the if statements to reduce cognitive load in general. e.g.
```suggestion
    if response.status_code != 200:
        raise ValueError(f"Invalid url {url}.")
    ## ...continue code...
```